plugins/lowering/src/main/java/org/qbicc/plugin/lowering/InvocationLoweringBasicBlockBuilder.java

```diff
@@ -212,25 +185,21 @@ public ValueHandle visit(ArrayList<Value> args, VirtualMethodElementHandle node)
             programModule.declareData(null, vtables.getName(), vtables.getType());
         }
         int index = dt.getVTableIndex(target);
-        Value typeId = fb.load(fb.instanceFieldOf(fb.decodeReference(node.getInstance()), CoreClasses.get(ctxt).getObjectTypeIdField()));
-        Value vtable = fb.load(elementOf(ctxt.getLiteralFactory().literalOf(dt.getVTablesGlobal()), typeId));
-        Value ptr = fb.load(memberOf(bitCast(vtable, info.getType().getPointer()), info.getType().getMember(index)));
-        return pointerHandle(ptr);
+        Value typeId = fb.load(instanceFieldOf(fb.decodeReference(reference), CoreClasses.get(ctxt).getObjectTypeIdField()));
+        Value vtable = fb.load(elementOf(lf.literalOf(dt.getVTablesGlobal()), typeId));
+        return fb.load(memberOf(bitCast(vtable, info.getType().getPointer()), info.getType().getMember(index)));
     }
 
     // Current implementation strategy is "searched itables" in the terminology of [Alpern et al 2001].
     @Override
-    public ValueHandle visit(ArrayList<Value> args, InterfaceMethodElementHandle node) {
+    public Value lookupInterfaceMethod(Value reference, InstanceMethodElement target) {
         final BasicBlockBuilder fb = getFirstBuilder();
-        // insert "this" and current thread
-        args.addAll(0, List.of(fb.load(fb.currentThread(), SingleUnshared), node.getInstance()));
-        final MethodElement target = node.getExecutable();
+        LiteralFactory lf = getLiteralFactory();
         DispatchTables dt = DispatchTables.get(ctxt);
         DispatchTables.ITableInfo info = dt.getITableInfo(target.getEnclosingType().load());
         if (info == null) {
             // No realized invocation targets are possible for this method!
-            MethodElement method = RuntimeMethodFinder.get(ctxt).getMethod("raiseIncompatibleClassChangeError");
-            throw new BlockEarlyTermination(fb.callNoReturn(staticMethod(method), List.of()));
+            return lf.nullLiteralOfType(target.getType().getPointer());
```

Comment:
I think this changes the exception from an IncompatibleClassChangeError to a NullPointerException.  Was the exception wrong before for this case (ie, we should have got the NPE before we got the ICCE)?

Reply:
If there are no realized targets then the invocation could not have been reachable with a non-`null` receiver, thus a previous `null` check on the receiver would always have fired. Complicating matters, since the method lookups are now schedulable nodes, they could be hoisted up to an earlier block before any such null check, thus we can't safely introduce flow control at this point because it might be too early, so producing a literal is the safest bet, and I considered `null` to be the best choice since that will cleanly segfault if such a lookup result is ever used.

Reply:
ok. makes sense.